### PR TITLE
add proper MRS integration

### LINKED ISSUE #2918: Owner request: Remove matrixrooms.info instance
Hi there,
I'm one of the etke.cc developers, and we operate MRS demo instance on matrixrooms.info / apicdn.matrixrooms.info

Please remove that **instance** from your code base - nobody contacted or notified us before its inclusion, and now we have to pay for the traffic your project generates.

Moreover, the integration metadata is completely invalid:

* [WikiData ID is wrong](https://www.wikidata.org/wiki/Q107565255)
* [Official API documentation is not provided](https://gitlab.com/etke.cc/mrs/api/-/blob/main/openapi.yml?ref_type=heads)
* It uses the official API
* Category should be instant messages

---

To avoid any misunderstanding, let me clarify some things.

1. You may want to integrate Matrix Rooms Search - self-hosted search engine over matrix federation: https://gitlab.com/etke.cc/mrs/api
2. You're supposed to host your own instance of MRS
3. The matrixrooms.info is a demo instance, and it is not intended for such integration
4. We pay for matrixrooms.info / apicdn.matrixrooms.info traffic from our own pockets and if it becomes unbearable, we can consider shutting it down.

### PR DESCRIPTION
**Pull request by aine-etke**
## What does this PR do?

Initial integration of MRS by @return42 is incorrect because it uses a demo instance of the project, and that demo instance is not intended for such integration. Plus, the initial integration promotes centralization on one single MRS instance, which is contrary to the MRS project's ideas.

## Related issues

Fixes #2918

### CODE REVIEW

**Comment by Bnyro** [comment]
https://github.com/searxng/searxng/issues/2918#issuecomment-1752157420

**Comment by Bnyro** [comment]
I disagree with removing the default instance url

**Comment by aine-etke** [comment]
There is no such thing as "default instance" in the case of MRS. The used URL is demo instance by project devs, inteded for demo purposes.

It's like saying "default instance of mailcow", or "default instance of vaultwarden", or "default instance of uptime kuma", or "default instance of wireguard"...

**Review by return42 — APPROVED** [review]
@etkecc We apologize for the inconvenience caused by our implementations.

I squashed some boilerplate stuff into your commit --> [diff](https://github.com/searxng/searxng/compare/b9f239b2a852d31fec7c3824b0b877a98cdeb63b..34b3ed949595e92bd54192ea9f97ddcb39cdf963)

With this PR we remove (comment out) the MRS search in the `settings.yml` configuration defaults.

If an instance owner wants to activate this engine, he has to configure it completely, if the base_url is not set, this will be detected during initialization and the engine remains inactive.

File: searx/engines/mrs.py
Comment (unixfox) [inline searx/engines/mrs.py:1]: This doesn't explain though to the user how he can host an instance for himself.

Comment (return42) [reply searx/engines/mrs.py:1]: To stop the glitches on apicdn.matrixrooms.info, I decided to merge this PR for now ... We can now implement improvements .. (for instance see my comment https://github.com/searxng/searxng/issues/2918#issuecomment-1752832630)

> However, this does not explain to the user how to host an instance for himself.

I think its not in the scope of a SearXNG engine to explain how a MRS instance is hosted / or did I misunderstood your expectations?

Comment (unixfox) [reply searx/engines/mrs.py:1]: Let's take I'm a simple user that do not know matrixrooms, I go to the link provided in the comments: https://docs.searxng.org/dev/engines/online/mrs.html

I see no actual explanations or external links on how to setup an instance of matrixrooms.

That's where we should improve the documentation.

**Comment by aine-etke** [comment]
Thank you for understanding!

However, @unixfox has a valid point about the hosting. I will be glad to help with MRS-related improvements. We can continue the discussions in the matrix room, if you don't mind

**Pull request merged**